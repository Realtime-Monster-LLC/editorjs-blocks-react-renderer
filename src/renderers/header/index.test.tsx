import React from 'react';
import { create } from 'react-test-renderer';
import Header, { HeaderBlockData } from '.';

describe('<Header />', () => {
  describe('when receives an header level 2 block', () => {
    const data: HeaderBlockData = {
      text: 'Editor.js',
      level: 2,
    };

    it('renders a <h2> tag', () => {
      expect(create(<Header data={data} />).toJSON()).toMatchSnapshot();
    });
  });

  describe('when receives a header block with className', () => {
    const data: HeaderBlockData = {
      text: 'Editor.js',
      level: 2,
    };

    it('renders a <h2 className>', () => {
      expect(create(<Header data={data} className={'header-block'} />).toJSON()).toMatchSnapshot();
    });
  });
});
