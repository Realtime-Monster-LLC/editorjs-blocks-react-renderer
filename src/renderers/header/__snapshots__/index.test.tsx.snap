// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> when receives a header block with className renders a <h2 className> 1`] = `
<h2
  className="header-block"
>
  Editor.js
</h2>
`;

exports[`<Header /> when receives an header level 2 block renders a <h2> tag 1`] = `
<h2>
  Editor.js
</h2>
`;
